src/components/SiderMenu/BaseMenu.tsx

```diff
@@ -15,39 +15,53 @@ const { SubMenu } = Menu
 //   icon: <Icon type="setting" />,
 const getIcon = icon => {
   if (typeof icon === 'string' && isUrl(icon)) {
-    return <img src={icon} alt="icon" className={styles.icon} />
+    return <img src={icon} alt='icon' className={styles.icon} />
   }
   if (typeof icon === 'string') {
     return <Icon type={icon} />
   }
   return icon
 }
 
-export default class BaseMenu extends PureComponent {
+export default function BaseMenu (props) {
+  const {
+    openKeys,
+    theme,
+    mode,
+    location: { pathname },
+    className,
+    collapsed,
+    handleOpenChange,
+    style,
+    menuData,
+    flatMenuKeys,
+    isMobile,
+    onCollapse
+  } = props
+
   /**
    * 获得菜单子节点
    * @memberof SiderMenu
    */
-  getNavMenuItems = (menusData, parent) => {
+  function getNavMenuItems (menusData, parent) {
     if (!menusData) {
       return []
     }
     return menusData
       .filter(item => item.name && !item.hideInMenu)
-      .map(item => this.getSubMenuOrItem(item, parent))
+      .map(item => getSubMenuOrItem(item, parent))
       .filter(item => item)
-  };
+  }
```

Comment:
let's move this function to the module scope and rename to `createNavMenuItems`

Reply:
done